scripts/extract-replaygain

```diff
@@ -2,22 +2,40 @@
 
 set -e
 
-FFMPEG=`which ffmpeg || echo ""`
+FFMPEG=`which ffmpeg 2>/dev/null || echo ""`
+FFPROBE=`which ffprobe 2>/dev/null || echo ""`
 
 if test -z "$FFMPEG"; then
   echo "ffmpeg must be installed"
   exit 1
 fi
 
-SED=`which gsed || echo ""`
+SED=`which gsed 2>/dev/null || echo ""`
 
 if test -z "$SED"; then
-  SED=`which sed || echo ""`
+  SED=`which sed 2>/dev/null || echo ""`
 fi
 
 if test -z "$SED"; then
-  echo "sed or gsed much be installed"
+  echo "sed or gsed must be installed"
   exit 1
 fi
 
-$FFMPEG -i "$1" -vn -filter replaygain -f null /dev/null 2>&1 | $SED -n -r 's/.* track_gain = ([-+]?[0-9]+\.[0-9]+ dB).*/\1/p'
+# check for replaygain metadata
+RG=`$FFPROBE "$1" 2>&1 | $SED -n -r 's/.* REPLAYGAIN_TRACK_GAIN: ([-+]?[0-9]+\.[0-9]+ dB).*/\1/p'`
+
+if [ -z "$RG" ]; then
+ # no replaygain metadata: compute it
+ RG=`$FFMPEG -i "$1" -vn -filter replaygain -f null /dev/null 2>&1 | $SED -n -r 's/.* track_gain = ([-+]?[0-9]+\.[0-9]+ dB).*/\1/p'`
+
+ # store it and keep file date
+ tmpdir=`mktemp -d`
```

Comment:
You should add a `trap` directive to the script to make sure that this temporary folder is always removed if the script exists prematurely..